### ISSUE TITLE
Export language as JAR

### ISSUE THREAD

**Issue by BramGrooteman** [graalvm/graal#231]
I have a question.
I build a subset of Java with Truffle in Eclipse which works nicely and now I want to export it as a JAR so I can use it somewhere else using the export function of Eclipse. But as soon as I do that and run it somewhere else it gives me:
`java.lang.IllegalStateException: No language for MIME type application/x-jsl found. Supported types: [application/x-tck]`
How do I register my language properly? Or what is the intended process for this?

**Comment by chumer** [comment]
You need to apply @TruffleLanguage.Registration on your class. Also make sure you enable annotation processing with the truffle-dsl-processor.jar when building. You can verify your language is registered properly if you've got a META-INF/truffle/language file in your jar file.

**Comment by BramGrooteman** [comment]
The thing is that when I run the code from within Eclipse, it does work properly. I can even loop over the map that I get from `engine.getLanguages();` in my main class and verify that my language is present. This is leading me to believe I registered properly. 
But the moment I use the Export function from Eclipse (`Export -> Runnable JAR file`) it loses the registration (So probably more of a Eclipse than a Truffle/Graal problem). Do I maybe have to put something in the `graalvm/language` folder? Any ideas?

**Comment by smarr** [comment]
Haven't used Eclipse to create jars. To automate things in CI, I am using a comparably simple ant build script: https://github.com/smarr/SOMns/blob/master/build.xml

SimpleLanguage uses Maven: https://github.com/graalvm/simplelanguage/blob/master/pom.xml

Perhaps either method is an option?

**Comment by BramGrooteman** [comment]
That did the job!
The problem was that when I first would use Maven it would not work due to the fact that I also used a JavaParser dependency that was being included so I got a `java.lang.NoClassDefFoundError` everytime I tried to run the jar. Since I did not know how to solve this I defaulted to using Eclipse. 
Got it working properly now by making sure the dependency is included. Thanks for helping me out!
